### ISSUE TITLE
pick a versioning scheme

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/toolset#14]
This project bundles at least 3 tools that have very different versioning and release cycles and we still want to make it easier for users to know what they should expect to get from inside.

Example:

* ansible 2.10.x
* molecule 3.1.x
* ansible-lint 4.3.x

What versioning should we pick?

Keep in mind that any release made by the above tools or even their dependencies would generate PRs to update our build requirements. When merged this will build and publish a `main` moving tag.

I do only expect a limited number of users to use the flexible tag as I suspect some would want to keep control over when to upgrade the toolset.

**Comment by ssbarnea** [comment]
I am inclined to pick something similar to [CalVer used by pip](https://github.com/pypa/pip/pull/5324) that would look like:

* `20.0` first release in 2020.
* `20.0.1` a bugfix release
* `20.1` second major release of 2020

And the rule for tagging would be simple: if any of the main components we bundle makes a major release we do bump our own release number. Everything else goes as a bugfix increment.

**Comment by gundalow** [comment]
Any reason not to use standard semvir starting at `1.0.0`?

**Comment by ssbarnea** [comment]
> Any reason not to use standard semvir starting at `1.0.0`?

The reason why I am inclined not leave the SemVer as a second option is because it does not give any indication about how old is a release by itself. Out of context it gives you no information. I also found confusing it with the other tools.

Keep in mind that the proposed CalVer is a flavor of SemVer, being totally compatible with it. We still fully respect the SemVer meaning.

The only difference is starting number, which gives some indication about when it happened.

Obviously that if on 2nd of January 2021 you need to make a patch release post a 20.1.0 one, you will just do it as `20.1.1`, even if the date when you do it is in 2021.

Basically the first number is incremented to match year of a major release. The first release that can break stuff in 2021, will likely be `21.0`, otherwise it would be something like`20.x.y+1`.

**Comment by ssbarnea** [comment]
Closing because I will delay making a decision until I get more feedback and the use of toolset project becomes more widespread.

**Comment by ssbarnea** [comment]
I decided to adopt versioning based on version of ansible package included. If we release more often we will use an extra digit to mark our updates.
